tests/testthat/test-assert.R

```diff
@@ -218,6 +218,12 @@ test_that("assertion failures are explained", {
   expect_snapshot(error = TRUE, vec_assert(data.frame(x = 1, y = 2), data.frame(x = "foo", y = 2)))
 })
 
+test_that("vec_assert() validates `size` (#1470)", {
+  expect_snapshot(error = TRUE, vec_assert(1, size = c(2, 3)))
```

Comment:
Can you please use the `expect_snapshot((expect_error()))` approach? This way the rlang print method gets actioned and we test the whole error prefix including call.